### PR TITLE
fix: prevent hang when process is overwritten

### PR DESCRIPTION
**Pull request by AriPerkkio**
- Fixes https://github.com/vitest-dev/vitest/issues/5390

### CODE REVIEW

**Review by hi-ogawa — APPROVED** [review]
Looks good!

File: src/entry/process.ts [hunk 1]
```diff
@@ -23,6 +23,9 @@ process.__tinypool_state__ = {
   workerId: Number(process.env.TINYPOOL_WORKER_ID),
 }
 
+const memoryUsage = process.memoryUsage.bind(process)
+const send = process.send!.bind(process)
```
Comment (Aslemammad) [inline src/entry/process.ts:27]: I assume this stabilizes the process and keeps the same reference, correct?

**Review by Aslemammad — APPROVED** [review]
LGTM! Feel free to merge.

File: src/entry/process.ts (hunk 1 above)
Comment (AriPerkkio) [reply src/entry/process.ts:27]: Yes, we will preserve the original `process` methods in the module scope. The file that is loaded later may overwrite globals:

https://github.com/tinylibs/tinypool/blob/6ca7bb480a9d0c54dac04eaab5e6523d300f1791/src/entry/process.ts#L35-L36

So basically:

```js
process.on("message", async (msg) => {
  const originalSend = process.send.bind(process);

  await import(msg.filename); // Contains globalThis.process = Mock;

  process.send(response);
  //      ^^^^ send is not a function

  originalSend(response);
  // ^^ works
})
```

`birpc` has similar implementation: https://github.com/antfu/birpc/blob/72510536cedb301953e0550d884667804d7268d0/src/index.ts#L145-L147

**Pull request merged**